cmd/srcd/cmd/components.go

```diff
@@ -17,49 +17,49 @@ package cmd
 import (
 	"context"
 	"fmt"
-	"log"
 	"os"
 	"strings"
 
-	"github.com/spf13/cobra"
 	"github.com/src-d/engine/components"
 	"github.com/src-d/engine/docker"
+
+	"gopkg.in/src-d/go-cli.v0"
+	"gopkg.in/src-d/go-log.v1"
 )
 
 // componentsCmd represents the components command
-var componentsCmd = &cobra.Command{
-	Use:   "components",
-	Short: "Manage source{d} components and their installations",
+type componentsCmd struct {
+	cli.PlainCommand `name:"components" short-description:"Manage source{d} components and their installations" long-description:"Manage source{d} components and their installations"`
 }
 
 // componentsListCmd represents the components list command
-var componentsListCmd = &cobra.Command{
-	Use:   "list",
-	Short: "List source{d} components",
-	RunE: func(cmd *cobra.Command, args []string) error {
-		allVersions, _ := cmd.Flags().GetBool("all")
+type componentsListCmd struct {
+	Command `name:"list" short-description:"List source{d} components" long-description:"List source{d} components"`
```

Comment:
maybe avoid repeating the same description? I'm not sure where `go-cli` uses `long-description`. If it's necessary for the nice output we can PR usage of short one if long is empty.

Reply:
Both are needed:
```$ srcd -h
Usage:
  srcd [OPTIONS] <command>

Available commands:
  version     Short goes here

$ srcd version -h
Usage:
  srcd [OPTIONS] version [version-OPTIONS]

Long goes here
```

I agree with your idea to use `short` when `long`  is not set. But I'd prefer if you could open an issue in https://github.com/src-d/go-cli instead of blocking this PR.